### PR TITLE
Remove netstandard1.3 from code base

### PR DESCRIPTION
**Pull request by jaredpar**
The remote debugger components moved to their minimum framework install
to 4.6.2 where it was previously 2.0. That means we can remove the uses
of `netstandard1.3` in our code base as they existed just to support
this specific case.

This means we can also remove our use of `net2.0` in the code base and
replace it with `net462`. That is a more involved changed though and
going to take that a piece at a time. Removing `netstandard1.3` is an
important step because presently it's blocking us from updating to the
latest System.Collections.Immutable as it dropped support for
`netstandard1.3`

### CODE REVIEW

**Comment by jaredpar** [comment]
@dotnet/roslyn-infrastructure, @tmat PTAL

File: src/Compilers/Test/Core/Generated.targets
Comment (Youssef1313) [inline src/Compilers/Test/Core/Generated.targets:156]: Looks like this is an old left over?

Comment (jaredpar) [reply src/Compilers/Test/Core/Generated.targets:156]: Actually all changes here are wrong, this is a generated file. This isn't us targeting `netstandard1.3` in our builds but rather testing the compilers ability to do builds against `netstandard1.3`

File: src/Dependencies/CodeAnalysis.Debugging/Microsoft.CodeAnalysis.Debugging.Package.csproj
Comment (tmat) [inline src/Dependencies/CodeAnalysis.Debugging/Microsoft.CodeAnalysis.Debugging.Package.csproj:5]: >netstandard1.3 [](http://example.com/codeflow?start=22&length=14)

These projects produce source package. Bumping the TFM means customers using the package won't be able to target netstandard1.3 anymore. Is that intentional?

Comment (jaredpar) [reply src/Dependencies/CodeAnalysis.Debugging/Microsoft.CodeAnalysis.Debugging.Package.csproj:5]: Hmm ... it wasn't intentional. At the same time is there any remaining realistic reason to support this? 

Unsure of the exact semantics of these packages. But given that everything is based on `netstandard2.0` now want to make sure we think supporting `netstandard1.3` here is still valuable.

Comment (tmat) [reply src/Dependencies/CodeAnalysis.Debugging/Microsoft.CodeAnalysis.Debugging.Package.csproj:5]: The policy for packages in general is to not update unless we need/want to use features of the newer platform. I am not against requiring `netstandard2.0` if it unlocks APIs that would be beneficial to use in the source.

**Comment by JoeRobich** [comment]
Test insertion - https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/376937 (microsoft)

File: src/Dependencies/CodeAnalysis.Debugging/Microsoft.CodeAnalysis.Debugging.Package.csproj
Comment (jaredpar) [reply src/Dependencies/CodeAnalysis.Debugging/Microsoft.CodeAnalysis.Debugging.Package.csproj:5]: Well we wouldn't be in this mess that required this PR if SCI / SRM had followed that policy 😄

Comment (jaredpar) [reply src/Dependencies/CodeAnalysis.Debugging/Microsoft.CodeAnalysis.Debugging.Package.csproj]: Moved back to `netstandard1.3` and it didn't impact any of our build scenarios that I can see so reverted this part of the change.

**Review by tmat — APPROVED** [review]

**Review by RikkiGibson — DISMISSED** [review]
Ensuring we don't merge until we have a passing VS test insertion.

**Comment by jaredpar** [comment]
@RikkiGibson verified offline with @333fred that the VS test run didn't reveal any issues hence think you are safe to remove your "request changes" now. If there are other checks you think we should run please let me know.

**Comment by jaredpar** [comment]
@dotnet/roslyn-infrastructure, @dotnet/roslyn-ide can i get a second review here? This is impacting the ability for VS Plat to upgrade S.C.I

**Review by 333fred — APPROVED** [review]
I believe we snapped for 17.2p2 at this point @RikkiGibson?

**Review by RikkiGibson — APPROVED** [review]

**Comment by RikkiGibson** [comment]
> I believe we snapped for 17.2p2 at this point

We hadn't at the time of this comment, but now we have.

**Comment by RikkiGibson** [comment]
Merge at will :)

**Comment by jaredpar** [comment]
Let's see how this goes ...

**Pull request merged**
